### PR TITLE
SW-1061 Add unit tests for notification email

### PR DESCRIPTION
**Pull request by sgrimm**
Add a test suite to confirm that the rendered email notification messages
include the expected content. This doesn't test the boilerplate text in the
template files since those are subject to change, but it attempts to verify
that the links are present and pointing to the right place and that the various
bits of dynamic content specified in the requirements docs are included.

It is still possible for someone to mess up an MJML template and cause the
notifications to look illegible, so when we revise notification content, we
will still want to do manual testing to confirm the results have the correct
appearance. But this set of tests will at least confirm that the templates
have no syntax errors and use the expected set of variable substitutions.

Writing this test suite turned up a bug: the notification for unknown
automations wasn't linking to the specific device on the monitoring dashboard.
That bug is fixed here.

### CODE REVIEW

**Comment by sgrimm** [comment]
Current dependencies on/for this PR:
* main
  * **PR #458** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/458" target="_blank">Graphite</a> 
    * **PR #459** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/459" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/459?utm_source=stack-comment).

**Comment by sgrimm** [comment]
This test has a lot of stubbing. I don't love that, but it's not clear to me that making this a database-backed test would be a net improvement; you'd replace the stub objects with a bunch of additional dependency wiring and of course it'd be slower.

File: src/main/kotlin/com/terraformation/backend/email/EmailNotificationService.kt [hunk 1]
```diff
@@ -128,12 +125,13 @@ class EmailNotificationService(
...
   fun on(event: UnknownAutomationTriggeredEvent) {
     val automation = automationStore.fetchOneById(event.automationId)
+    val devicesRow = automation.deviceId?.let { deviceStore.fetchOneById(it) }
     val facility = facilityStore.fetchOneById(automation.facilityId)
     val organizationId =
         parentStore.getOrganizationId(facility.id) ?: throw FacilityNotFoundException(facility.id)
 
     val facilityMonitoringUrl =
-        webAppUrls.fullFacilityMonitoring(organizationId, facility.id).toString()
+        webAppUrls.fullFacilityMonitoring(organizationId, facility.id, devicesRow).toString()
```
Comment (karthikbtf) [inline src/main/kotlin/com/terraformation/backend/email/EmailNotificationService.kt:134]: good catch with the tests

**Comment by karthikbtf** [comment]
@sgrimm this file has conflicts with the previously merged diff (though i understand the changes are the same).

**Pull request merged**